### ISSUE TITLE
[Bug]: Azure AD authentication fails with a working configuration from v0.21.2 -> v0.22.0+

### ISSUE THREAD

**Issue by ericj-msft** [gravitl/netmaker#2836]
### Contact Details

eric@loopr.ai

### What happened?

Using the same working configuration since prior to v0.21.2 and docker containers, changing the SERVER_IMAGE_TAG in netmaker.env to v0.22.0 or later will no longer allow SSO logins, claiming that "Your Netmaker server does not have OAuth configured.", despite the same configuration working with v0.21.2 and earlier if SERVER_IMAGE_TAG is set to these versions.

### Version

v0.23.0

### What OS are you using?

Linux

### Relevant log output

```shell
[netmaker] 2024-03-01 06:35:31 processed request error: unauthorized
[netmaker] 2024-03-01 06:35:52 error retrieving oauth state: no result found
[netmaker] 2024-03-01 06:35:52 error when getting user info from azure: invalid oauth state
```

### Contributing guidelines

- [X] Yes, I did.

**Comment by ericj-msft** [comment]
OAuth section from `netmaker.env`:

```
###
#
# OAuth section
#
###
# "<azure-ad|github|google|oidc>"
AUTH_PROVIDER=azure-ad
# "<client id of your oauth provider>"
CLIENT_ID=_<REDACTED>_
# "<client secret of your oauth provider>"
CLIENT_SECRET=_<REDACTED>_
SERVER_HTTP_HOST=api._<REDACTED>_
# "https://dashboard.<netmaker base domain>"
FRONTEND_URL=https://dashboard._<REDACTED>_
# "<only for azure, you may optionally specify the tenant for the OAuth>"
AZURE_TENANT=_<REDACTED>_
# https://oidc.yourprovider.com - URL of oidc provider
OIDC_ISSUER=https://login.microsoftonline.com/_<SAME AS AZURE_TENANT>_/v2.0/
EXPORTER_API_PORT=8085
JWT_VALIDITY_DURATION=43200
RAC_AUTO_DISABLE=true
```

**Comment by abhishek9686** [comment]
there were no changes to Oauth in recent releases, will check this on our side to determine if there is any possible issue and any configuration needs to updated

**Comment by abhishek9686** [comment]
@ericj-msft patched the v0.23.0 with a fix for the azure-ad, can you try again by pulling latest image of v0.23.0
